### PR TITLE
janitor modsuit space cleaner mister

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request
2024-11-17 10_33_43-Window

generates 2u of space cleaner per second if active
it shoots space cleaner
janitor ert gets it

## Why It's Good For The Game

at long last janitors get some actually useful modsuit module

## Changelog
:cl:
add: janitor modsuit space cleaner mister module
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
is this added to ERT jannies?

**Comment by mc-oofert** [comment]
> is this added to ERT jannies?

no(t yet)

**Comment by ju45he** [comment]
I think it could use more purple.

File: code/modules/mod/mod_types.dm [hunk 1]
```diff
@@ -495,7 +495,7 @@
 
 /obj/item/mod/control/pre_equipped/responsory/janitor
 	insignia_type = /obj/item/mod/module/insignia/janitor
-	additional_module = /obj/item/mod/module/noslip
+	additional_module = list(/obj/item/mod/module/noslip, /obj/item/mod/module/mister/cleaner)
```
Comment (ZephyrTFA) [inline code/modules/mod/mod_types.dm:498]: does this support being a list? if yes you should change it to be called `additional_modules`

File: code/modules/mod/modules/modules_service.dm [hunk 2]
```diff
@@ -91,3 +91,21 @@
...
+/obj/item/mod/module/mister/cleaner
+	name = "MOD janitorial mister module"
+	desc = "An space cleaner mister, able to clean up messes quickly. Synthesizes its own supply over time (if active)."
+	device = /obj/item/reagent_containers/spray/mister/janitor
+	volume = 100
+	active_power_cost = DEFAULT_CHARGE_DRAIN
+
+/obj/item/mod/module/mister/cleaner/Initialize(mapload)
+	. = ..()
+	reagents.flags = AMOUNT_VISIBLE
```
Comment (ZephyrTFA) [inline code/modules/mod/modules/modules_service.dm:105]: are you meaning to override entirely or do you want to `|=` here?

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/modules/mod/modules/modules_service.dm (hunk 2 above)
Comment (mc-oofert) [reply code/modules/mod/modules/modules_service.dm:105]: Override

**Comment by Dawnseer** [comment]
Very nice

**Pull request merged**